contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,866 @@
+            : valInfo.commission.rate - commissionRate;
+        require(changeRate <= valInfo.commission.maxChangeRate, "INVALID_COMMISSION_RATE");
+
+        valInfo.commission.rate = commissionRate;
+        valInfo.updateTime = block.timestamp;
+
+        emit CommissionRateEdited(operatorAddress, commissionRate);
+    }
+
+    function editDescription(Description calldata description) external onlyInitialized validatorExist(msg.sender) {
+        address operatorAddress = msg.sender;
+        Validator storage valInfo = _validators[operatorAddress];
+        require(valInfo.updateTime + 1 days <= block.timestamp, "UPDATE_TOO_FREQUENTLY");
+        require(_checkMoniker(description.moniker), "INVALID_MONIKER");
+
+        valInfo.description = description;
+        valInfo.updateTime = block.timestamp;
+
+        emit DescriptionEdited(operatorAddress);
+    }
+
+    function editVoteAddress(
+        bytes calldata newVoteAddress,
+        bytes calldata blsProof
+    ) external onlyInitialized validatorExist(msg.sender) {
+        address operatorAddress = msg.sender;
+        Validator storage valInfo = _validators[operatorAddress];
+        require(valInfo.updateTime + 1 days <= block.timestamp, "UPDATE_TOO_FREQUENTLY");
+        require(_voteToOperator[newVoteAddress] == address(0), "DUPLICATE_VOTE_ADDRESS");
+        require(_checkVoteAddress(newVoteAddress, blsProof), "INVALID_VOTE_ADDRESS");
+
+        bytes memory oldVoteAddress = valInfo.voteAddress;
+        delete _voteToOperator[oldVoteAddress];
+        _voteToOperator[newVoteAddress] = operatorAddress;
+
+        valInfo.voteAddress = newVoteAddress;
+        valInfo.updateTime = block.timestamp;
+
+        emit VoteAddressEdited(operatorAddress, newVoteAddress);
+    }
+
+    function unjail(address operatorAddress) public onlyInitialized validatorExist(operatorAddress) {
+        Validator storage valInfo = _validators[operatorAddress];
+        require(valInfo.jailed, "NOT_JAILED");
+
+        address pool = valInfo.poolModule;
+        require(IStakePool(pool).getSelfDelegationBNB() >= minSelfDelegationBNB, "NOT_ENOUGH_SELF_DELEGATION");
+        require(valInfo.jailUntil <= block.timestamp, "STILL_JAILED");
+
+        valInfo.jailed = false;
+        emit ValidatorUnjailed(operatorAddress);
+    }
+
+    function delegate(address operatorAddress)
+        external
+        payable
+        onlyInitialized
+        validatorExist(operatorAddress)
+        whenNotPaused
+    {
+        uint256 bnbAmount = msg.value;
+        require(bnbAmount >= minDelegationBNBChange, "INVALID_DELEGATION_AMOUNT");
+
+        address delegator = msg.sender;
+        Validator memory valInfo = _validators[operatorAddress];
+        address pool = valInfo.poolModule;
+        require(_checkPoolImplementation(pool), "IMPLEMENTATION_NOT_MATCH");
+
+        if (valInfo.jailed || IStakePool(pool).getSelfDelegationBNB() < minSelfDelegationBNB) {
```

Comment:
Is that possible the second condition is ture, while it is not jailed?

Reply:
No. Updated